### ISSUE TITLE
Rust derive(Clone) does not seem to work for more complicated examples

### ISSUE THREAD

**Issue by uzytkownik** [rust-lang/rust#48974]
The code below is what I would expect the `Clone` to look like based on [`Clone` for `Addr`](https://actix.github.io/actix/actix/struct.Addr.html#impl-Clone). However while Rust doesn't emit doesn't emit any errors during compilation it does create some other implementation which is more restrictive and didn't worked in my code.

```rust
//#[derive(Clone)]
struct ActorData<T: Actor> {
	addr: Addr<Syn, T>
}

impl<T: Actor> Clone for ActorData<T> {
	fn clone(&self) -> ActorData<T> {
		ActorData {
			addr: self.addr.clone()
		}
	}
	fn clone_from(&mut self, source: &Self) {
		self.addr.clone_from(&source.addr)
	}
}
```

**Comment by oli-obk** [comment]
Duplicate of https://github.com/rust-lang/rust/issues/26925

TLDR: You can't do this with derive because the derive impl adds a `Clone` bound to `T`

**Comment by Ixrec** [comment]
For completeness: https://github.com/rust-lang/rfcs/pull/2353 is an RFC that would help with cases like this.
